### ISSUE TITLE
Is the creation of embeddings not determinisctic?

### ISSUE THREAD

**Issue by Moiddes** [flairNLP/flair#1578]
I'm working on a project about document/paragraph similarity and I wanted to incorporate flair embeddings in my research, however results have been worse than expected, so I started bug searching in my code. I'm using 
```
embedding = BertEmbeddings(layers='-1')
document_embeddings = DocumentPoolEmbeddings([embedding], fine_tune_mode='nonlinear')
```
as my model for creating document embeddings using flair right now. (Whether these are good or bad representations of documents for assessing similarity is another question, hence the research)

I've written a generator function for generating embeddings from strings:

```
def generate_embeddings(docs, batch_size, model=document_embeddings, offset=0):
    rest = len(docs) % batch_size
    for i in range(0, len(docs) - rest, batch_size):
        sentences = [Sentence(sentence)for sentence in docs[i:i + batch_size]]
        try:
            model.embed(sentences)
            print(f'successfully embedded sentences {offset + i} to {offset + i + batch_size-1}')
            yield 1, [sentence.get_embedding().detach().cpu().numpy() for sentence in sentences]
        except RuntimeError:
            print(f'could not embed sentences with index {offset + i} '
                  f'to {offset + i + batch_size-1}\nstoring in failed index list')
            yield 0, (offset + i, offset + i + batch_size-1)
    if rest:
        sentences = [Sentence(sentence)for sentence in docs[-rest:]]
        try:
            model.embed(sentences)
            print(f'successfully embedded sentences from {len(docs) + offset - rest} to the end')
            yield 2, [sentence.get_embedding().detach().cpu().numpy() for sentence in sentences]
        except RuntimeError:
            yield 0, (len(docs) - rest, 0)

```

Now, when I create embeddings of the same inputs (which are sentences, not long documents) more than once, I get different representations of my inputs. 

so:
**Is the creation of embeddings not determinisctic?**

Here are two pictures which show the first elements of vectors, which are different, but were generated with the same text as input:
![Array Vectors from 0 to 3 before normalization](https://user-images.githubusercontent.com/1322289/81058057-f0589b00-8ecd-11ea-9ad1-f1f49b5ac0f1.png)
![new Array Vectors (0 to 3) are different?](https://user-images.githubusercontent.com/1322289/81058061-f3ec2200-8ecd-11ea-8b00-a91a8e3c1d6e.png)

I expected : same input -> same embeddings. However, that seems to not be the case.

**Comment by alanakbik** [comment]
Hello @Moiddes one thing first is that `DocumentPoolEmbeddings` with `fine_tune_mode='nonlinear'` are a _trainable_ embedding, meaning that the trainable parts get randomly initialized when you create the object and need to be trained in order to make sense. 

If you want to use a non-trainable embedding, you should use `DocumentPoolEmbeddings` with `fine_tune_mode='none'`. In this case, it is always a simple pool over word embeddings and this embedding should always be the same. Could you try with `fine_tune_mode='none'` to see if the error goes away?

**Comment by Moiddes** [comment]
Thanks very much! I indeed always get the same embeddings with `fine_tune_mode='none'`. I chose `fine_tune_mode='nonlinear'`, since the tutorial said to choose it, if embeddings are not task trained (yet). But as I don't intend to task-train the embeddings, I shouldn't fine-tune to begin with. Thanks for the clarification!
